### PR TITLE
Dynamic footer

### PR DESCRIPTION
**Pull request by RadekVana**
complete footer rewrite
also adds optional relative positioning into windows
new footer support changing displayed items at runtime and also dynamic positioning (on item value change)
BFW-1872
BFW-1807

### CODE REVIEW

File: src/gui/footer/footer_items_heaters.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    //replace front zeroes with spaces
+    for (int i = index; i <= int(index) + 1; ++i) {
+        if (buff[i] == '0')
+            buff[i] = ' ';
+        else
+            return;
+    }
+}
```
Comment (dragomirecky) [inline src/gui/footer/footer_items_heaters.cpp:88]: Did you have a performance issue with snprintf?

File: src/gui/footer/footer_item_printspeed.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+int FooterItemSpeed::static_readValue() {
+    return marlin_vars()->print_speed;
+}
+
+string_view_utf8 FooterItemSpeed::static_makeView(int value) {
+    static char buff[5]; //max 999%
+    int value_to_print = std::clamp(value, 1, 999);
+    snprintf(buff, sizeof(buff), "%d%%", value_to_print);
+    return string_view_utf8::MakeRAM((const uint8_t *)buff);
```
[the remark this answers is not in the document]
Comment (RadekVana) [reply src/gui/footer/footer_item_printspeed.cpp:25]: max speed on mini is 255

[the remark this answers is not in the document]
Comment (RadekVana) [reply src/gui/footer/footer_item_printspeed.cpp:25]: so 999 is just print size limit

File: src/gui/footer/footer_item_live_z.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            uint i = uval % modulo;
+            if (i == 0)
+                break;
+
+            *ptr = '0' + i / (modulo / 10);
+            ++ptr;
+        }
+
+        *ptr = '\0';
+    }
```
[the remark this answers is not in the document]
Comment (RadekVana) [reply src/gui/footer/footer_item_live_z.cpp]: fixed

File: src/gui/footer/footer_items_heaters.cpp (hunk 1 above)
Comment (RadekVana) [reply src/gui/footer/footer_items_heaters.cpp]: fixed

**Review by dragomirecky — APPROVED** [review]

**Pull request merged**